lib/dor/text_extraction/sqs_watcher.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Dor
+  module TextExtraction
+    # Watch an SQS queue for messages by long polling, allowing the consumer
+    # of this class to specify the queue name and the processing logic.
+    class SqsWatcher
+      attr_reader :logger, :queue_url, :role_arn, :visibility_timeout
+
+      def initialize(queue_url:, role_arn: nil, logger: Logger.new($stdout), visibility_timeout: nil)
+        @queue_url = queue_url
+        @role_arn = role_arn
+        @logger = logger
+        @visibility_timeout = visibility_timeout || default_visibility_timeout
+      end
+
+      def poll
+        logger.info("Starting indefinite long polling of #{queue_url}")
```

Comment:
I think `infinite` is what you meant here? :-)

Reply:
maybe I'll poll for now, maybe I won't ... got a problem with that?

Reply:
ha, i had actually thought about a message like `"Polling forever on #{queue_url}"`, but then i figured we actually know that it'll run for a while, but that it'll be interrupted by deploys and outages and such 🙂 

maybe it will actually run for a definite amount of time if i deploy it before addressing the token expiration concern 😆